modules/adserveradtargetBidAdapter.js

```diff
@@ -0,0 +1,408 @@
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { deepAccess, isFn, isStr, isNumber, isArray, isEmpty, isPlainObject, generateUUID, logWarn } from '../src/utils.js';
+import { config } from '../src/config.js';
+import { hasPurpose1Consent } from '../src/utils/gpdr.js';
+
+const INTEGRATION_METHOD = 'prebid.js';
+const BIDDER_CODE = 'adserveradtarget';
```

Comment:
shorter bidder codes are preferred. Codes over 8 characters get truncated for some targeting keys to gam, eg hb_cache_id_adserveradtarget would truncate to hb_cache_id_adserver.